File: client/src/demos/code-execution-demo.py
# Python Code Execution Demo
# This file contains sample code to demonstrate the output feature

# Simple variable declarations and operations
greeting = "Hello, World!"
print(greeting)

# Working with lists
numbers = [1, 2, 3, 4, 5]
print("Original list:", numbers)

# List comprehension
doubled = [num * 2 for num in numbers]
print("Doubled list:", doubled)

# Dictionary manipulation
user = {
  "name": "John Doe",
  "age": 30,
  "occupation": "Developer",
  "skills": ["Python", "Django", "Flask"]
}
print("User dictionary:", user)

# Conditional logic
if user["age"] > 25:
  print(f"{user['name']} is over 25 years old")
else:
  print(f"{user['name']} is 25 or younger")

# Function definition and execution
def calculate_area(width, height):
  return width * height

rectangle_area = calculate_area(5, 3)
print(f"Rectangle area: {rectangle_area} square units")

# Error handling example
try:
  # This will cause an error
  result = undefined_variable + 10
  print(result)
except Exception as error:
  print("Error caught:", str(error))

# List operations
fruits = ["apple", "banana", "cherry"]
fruits.append("orange")
print("Fruits list after append:", fruits)

# Working with tuples
coordinates = (10, 20)
x, y = coordinates
print(f"X: {x}, Y: {y}")

print("End of demo script")

File: client/src/demos/code-execution-demo.js
// JavaScript Code Execution Demo
// This file contains sample code to demonstrate the output feature

// Simple variable declarations and operations
const greeting = "Hello, World!";
console.log(greeting);

// Working with arrays
const numbers = [1, 2, 3, 4, 5];
console.log("Original array:", numbers);

// Array methods
const doubled = numbers.map(num => num * 2);
console.log("Doubled array:", doubled);

// Object manipulation
const user = {
  name: "John Doe",
  age: 30,
  occupation: "Developer",
  skills: ["JavaScript", "React", "Node.js"]
};
console.log("User object:", user);

// Conditional logic
if (user.age > 25) {
  console.log(`${user.name} is over 25 years old`);
} else {
  console.log(`${user.name} is 25 or younger`);
}

// Function definition and execution
function calculateArea(width, height) {
  return width * height;
}

const rectangleArea = calculateArea(5, 3);
console.log(`Rectangle area: ${rectangleArea} square units`);

// Error handling example
try {
  // This will cause an error
  const result = undefinedVariable + 10;
  console.log(result);
} catch (error) {
  console.log("Error caught:", error.message);
}

// Asynchronous code example (output will be shown after synchronous code)
setTimeout(() => {
  console.log("This message appears after a delay");
}, 1000);

console.log("End of demo script");